src/wmtk/attribute/AttributeScopeStack.cpp

```diff
@@ -93,6 +101,19 @@ AttributeScope<T> const* AttributeScopeStack<T>::get_checkpoint(long index) cons
         return m_checkpoints.at(index);
     }
 }
+template <typename T>
+void AttributeScopeStack<T>::change_to_parent_scope()
+{
+    assert(!empty());
+    m_current = m_current->parent();
```

Comment:
assert that m_current->parent() != nullptr?
can m_current ever be nullptr / what is the spec like?

Reply:
I assume in the accessor that `m_current->parent()` can be a nullptr. In that case I am accessing the base instead of the cache.